include/zephyr/mgmt/hawkbit.h

```diff
@@ -68,6 +68,26 @@ enum hawkbit_response hawkbit_probe(void);
  */
 void hawkbit_reboot(void);
 
+/**
+ * @brief Callback to get the device identity.
+ *
+ * @param id Pointer to the buffer to store the device identity
+ * @param id_max_len The maximum length of the buffer
+ */
+typedef bool (*hawkbit_get_device_identity_cb_handler_t)(char *id, int id_max_len);
+
+/**
+ * @brief Set the device identity callback.
+ *
+ * @details This function is used to set a custom device identity callback.
+ *
+ * @param cb The callback function.
+ *
+ * @retval 0 on success.
+ * @retval -EINVAL if the callback is NULL.
```

Comment:
To match with what we have in `hawkbit_probe` above:

```suggestion
 * @return 0 on success.
 * @return -EINVAL if the callback is NULL.
```